src/aws-manager.js

```diff
@@ -31,18 +89,32 @@ class AwsManager {
     assert(pubKey);
     assert(maxInstanceLife);
     assert(influx);
+
     this.ec2 = ec2;
     this.provisionerId = provisionerId;
     this.keyPrefix = keyPrefix;
     this.pubKey = pubKey;
     this.maxInstanceLife = maxInstanceLife;
     this.influx = influx;
+
+    // Known keypairs are tracked so that we don't have to retreive the list of
+    // all known key pairs on every iteration.
     this.__knownKeyPairs = [];
+
+    // The responses of the EC2 Api's view of state
     this.__apiState = {
       instances: [],
       requests: [],
     };
+
+    // We store the state of the previous iteration so that we can do
+    // comparisons to see which instances have changed state
     this.__previousApiState = this.__apiState;
+
+    // Internal state is state which we know exists but has not yet been
+    // reflected in the EC2 api.  This list is used to track the spot requests
+    // which we have submitted but which do not yet show up in the ec2 api
+    // calls which list the spot requests
     this.__internalState = [];
```

Comment:
yah.  there will be a refactor, and when it happens the changes will be large and far reaching.  this name likely won't survive.  `_submittedSpotRequests` isn't really right, since that's all spot requests, not just these ones.  The most specific name would be `_submittedButNotYetVisibleToApiSpotRequests` and I think that's for sure way too long!

We already have one list of all live instances and all live spot requests.  The dead instances and requests aren't really all that important outside of figuring out termination reasons, so no need to mess up every single other function with and extra `.filter` call, imo.